__tests__/unique-names-generator.test.ts

```diff
@@ -185,4 +196,26 @@ describe('randomNameGenerator', () => {
     // Assert
     expect(() => expected()).toThrowErrorMatchingSnapshot();
   });
+
+  it('should use dictionaries in random order', () => {
+    const adjectives = ['Adjective1', 'Adjective2'];
+    const colors = ['Color1', 'Color2'];
+    const subjects = ['Animal1', 'Animal2'];
+
+    // Arrange
+    const config: Config = {
+      dictionaries: [adjectives, colors, subjects],
+      length: 2,
+      separator: '_',
+      random: true,
+    };
+
+    // Act
+    const uniqueNamesGenerator = new UniqueNamesGenerator(config);
+    const expected = () => uniqueNamesGenerator.generate();
+
+    // Assert
+    // expect(() => expected()).toThrowErrorMatchingSnapshot();
```

Comment:
@andreasonny83 Can you help me out here please. how would i test it?

Reply:
Hey @erdahuja . Can you please rename `random` to `shuffleDictionaries` instead.
Also, please don't commit the `yarn.lock`.

Reply:
I would probably test the logic in this way:
```
  it('should always use the same dictionaries when shuffleDictionaries is disabled', () => {
    // Arrange
    const dictionary1 = ['dictionary1'];
    const dictionary2 = ['dictionary2'];
    const dictionary3 = ['dictionary3'];
    const dictionary4 = ['dictionary4'];
    const dictionary5 = ['dictionary5'];

    const config: Config = {
      dictionaries: [dictionary1, dictionary2, dictionary3, dictionary4, dictionary5],
      length: 2,
      separator: '_',
      shuffleDictionaries: false,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);

    const res1 = uniqueNamesGenerator.generate();
    const res2 = uniqueNamesGenerator.generate();
    const res3 = uniqueNamesGenerator.generate();

    // Assert
    expect(res1).toEqual(res2);
    expect(res1).toEqual(res3);
  });

  it('should shuffle the provided dictionaries when shuffleDictionaries is enabled', () => {
    // Arrange
    const dictionary1 = ['dictionary1'];
    const dictionary2 = ['dictionary2'];
    const dictionary3 = ['dictionary3'];
    const dictionary4 = ['dictionary4'];
    const dictionary5 = ['dictionary5'];

    const config: Config = {
      dictionaries: [dictionary1, dictionary2, dictionary3, dictionary4, dictionary5],
      length: 2,
      separator: '_',
      shuffleDictionaries: true,
    };

    // Act
    const uniqueNamesGenerator = new UniqueNamesGenerator(config);

    const res1 = uniqueNamesGenerator.generate();
    const res2 = uniqueNamesGenerator.generate();

    // Assert
    expect(res1).not.toEqual(res2);
  });
```